app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/UserDictionaryAddWordContents.java

```diff
@@ -0,0 +1,363 @@
+            // Special treatment for the known languages with _zz and _ZZ types
+            } else if (localeString.endsWith("_zz") || localeString.endsWith("_ZZ")) {
+                final int resId = context.getResources().getIdentifier("subtype_"+localeString, "string", context.getPackageName());
+                mDescription = context.getString(resId);
+            } else {
+                mDescription = LocaleUtils.constructLocaleFromString(localeString).getDisplayName();
+            }
+        }
+        @Override
+        public String toString() {
+            return mDescription;
+        }
+        public String getLocaleString() {
+            return mLocaleString;
+        }
+
+    }
+
+    private static void addLocaleDisplayNameToList(final Context context,
+            final ArrayList<LocaleRenderer> list, final String locale) {
+        if (null != locale) {
+            list.add(new LocaleRenderer(context, locale));
+        }
+    }
+
+    // Helper method to get the list of locales and subtypes to display for this word
+    public ArrayList<LocaleRenderer> getLocalesList(final Activity activity) {
+
+        final SharedPreferences prefs = DeviceProtectedUtils.getSharedPreferences(activity.getApplicationContext());
+        final boolean localeSystemOnly = prefs.getBoolean(Settings.PREF_USE_SYSTEM_LOCALES, true);
+        final ArrayList<LocaleRenderer> localesList = new ArrayList<>();
+
+        // List of main language
+        final List<InputMethodSubtype> enabledMainSubtype = SubtypeSettingsKt.getEnabledSubtypes(prefs, true);
+        // List of system language
+        final List<Locale> enabledSystemLocale = SubtypeSettingsKt.getSystemLocales();
+        // List of user dictionary
+        final TreeSet<String> userDictionaryList = UserDictionaryListFragment.getUserDictionaryLocalesSet(activity);
+        // To combine lists
+        final TreeSet<String> languageList = new TreeSet<>(String::compareToIgnoreCase);
+
+        // Add languages from the user dictionary list
+        // (local system languages and dictionary languages in which a word is already saved)
+        if (userDictionaryList != null) {
+            languageList.addAll(userDictionaryList);
+        }
+
+        // mLocale is removed from the language list as it will be added to the top of the list
+        languageList.remove(mLocale);
+
+        // Add the main language selected in the "Language and Layouts" setting except "No language"
+        for (InputMethodSubtype mainSubtype : enabledMainSubtype) {
+            Locale mainLocale = LocaleUtils.constructLocaleFromString(mainSubtype.getLocale());
+            String mainLocaleString = mainLocale.toString();
+            if (userDictionaryList != null && !userDictionaryList.contains(mainLocale.toString())
+                    && !mainLocaleString.equals(mLocale) && !mainLocaleString.equals("zz")) {
+                languageList.add(mainLocaleString);
```

Comment:
You create the locale from a string, but then never use the locale, only `mainLocale.toString()`

Reply:
Change made with commit 0582df5